taskfiles/deps.yaml

```
@@ -0,0 +1,77 @@
+version: "3"
+
+includes:
+  utils: "../tools/yscope-dev-utils/exports/taskfiles/utils/utils.yaml"
+
+vars:
+  G_DEPS_DIR: "{{.G_BUILD_DIR}}/deps"
+
+  # This path must be kept in-sync with its usage in CMakeLists.txt and examples/CMakeLists.txt.
+  G_DEPS_CMAKE_SETTINGS_DIR: "{{.G_DEPS_DIR}}/cmake-settings"
+
+tasks:
+  default:
+    deps:
+      - "all"
+
+  all:
+    run: "once"
+    cmds:
+      - task: "utils:cmake:install-deps-and-generate-settings"
+        vars:
+          CMAKE_SETTINGS_DIR: "{{.G_DEPS_CMAKE_SETTINGS_DIR}}"
+          DEP_TASK: "all-parallel"
+
+  all-parallel:
+    internal: true
+    run: "once"
+    deps:
+      - "Catch2"
+      - "fmt"
+      - "Microsoft.GSL"
+
+  Catch2:
+    internal: true
+    run: "once"
+    cmds:
+      - task: "utils:cmake:install-remote-tar"
+        vars:
+          CMAKE_SETTINGS_DIR: "{{.G_DEPS_CMAKE_SETTINGS_DIR}}"
+          FILE_SHA256: "18b3f70ac80fccc340d8c6ff0f339b2ae64944782f8d2fca2bd705cf47cadb79"
+          GEN_ARGS:
+            - "-DCMAKE_CXX_STANDARD=20"
+          NAME: "Catch2"
```

```suggestion
          NAME: "{{.LIB_NAME}}"
```

Ditto for other tasks.